sympy/polys/monomials.py

```diff
@@ -44,18 +48,20 @@ def itermonomials(variables, degree):
         >>> sorted(itermonomials([x, y], 3), key=monomial_key('grlex', [y, x]))
         [1, x, y, x**2, x*y, y**2, x**3, x**2*y, x*y**2, y**3]
 
-    """
-    if not variables:
-        return set([S.One])
-    else:
-        x, tail = variables[0], variables[1:]
-
-        monoms = itermonomials(tail, degree)
+        >>> a, b = symbols('a, b', commutative=False)
+        >>> itermonomials([a, b, x], 2)
+        {1, a, a**2, b, b**2, x, x**2, a*b, b*a, x*a, x*b}
 
-        for i in range(1, degree + 1):
-            monoms |= set([ x**i * m for m in itermonomials(tail, degree - i) ])
 
-        return monoms
+    """
+    if not variables or degree < 1:
+        return {S(1)}
```

Comment:
Maybe the empty set should be returned when `degree < 0`.

Reply:
That sounds reasonable. Okay to change behavior from previous implementation?

Reply:
I think it is ok in this case. There are no monomials of degree < 0.